File: edge-cases/java/framework-annotations/Sample.java
package edge.annotations;

@Route("csv/import")
public final class Sample {
    @RequestMapping(method = "POST", consumes = "text/csv")
    public String upload(@Param("payload") String payload) {
        return payload.trim();
    }
}

@interface Route {
    String value();
}

@interface RequestMapping {
    String method();
    String consumes();
}

@interface Param {
    String value();
}

File: fixtures/edge-static-j2k/framework-annotations/Sample.kt
package edge.annotations

@Route("csv/import")
class Sample {
    @RequestMapping(method = "POST", consumes = "text/csv")
    fun upload(@Param("payload") payload: String): String {
        return payload.trim { it <= ' ' }
    }
}

internal annotation class Route(val value: String)

internal annotation class RequestMapping(val method: String, val consumes: String)

internal annotation class Param(val value: String)
